packages/vitest/src/node/mocker.ts

```diff
@@ -91,7 +91,7 @@ export class VitestMocker {
     const path = await this.options.resolveId(id, importer)
     return {
       path: normalizeId(path?.id || id),
-      external: path?.external ?? path?.id.includes('/node_modules/') ? id : null,
+      external: path?.external || path?.id.includes('/node_modules/') ? id : null,
```

Comment:
I'm not sure that this is correct. Theoretically `node_modules` file can be marked as not external.

Reply:
This is only for deciding when to store `mocks`. If it's `node_modules`, mocks should be placed in root.

Reply:
I've removed `path?.external` tho, don't know why I added it in this PR, this place should only check for `node_modules`